src/components/Tooltip.vue

```diff
@@ -34,217 +34,343 @@ import { Component, Prop, Vue, Watch } from 'vue-property-decorator';
+        TOP = 'top',
+        BOTTOM = 'bottom',
+    }
+
+    export enum Themes {
+        NORMAL = 'normal',
+        INVERSE = 'inverse',
     }
 }
+
+export default Tooltip;
 </script>
 
 <style scoped>
     .tooltip {
-        display: block;
+        display: inline-block;
         position: relative;
         line-height: 1;
     }
 
-    .tooltip > a {
+    .trigger {
         position: relative;
         display: block;
+        text-decoration: none;
+        outline: none;
+        cursor: default;
     }
 
-    .tooltip > a >>> svg {
+    .trigger >>> svg {
         display: block;
     }
 
-    .tooltip > a::after {
+    .trigger::after {
         opacity: 0;
         content: '';
         display: block;
         position: absolute;
-        left: calc(50% - 1rem);
-        border-width: 1rem;
-        border-style: solid;
-        transition: bottom .2s ease, top .2s ease, opacity .3s ease;
+        width: 2.25rem;
+        height: 2rem;
+        left: calc(50% - 1.125rem);
+        background: var(--nimiq-blue-bg-darkened);
+        mask-image: url('data:image/svg+xml,<svg viewBox="0 0 18 16" xmlns="http://www.w3.org/2000/svg"><path d="M9 7.12c-.47 0-.93.2-1.23.64L3.2 14.29A4 4 0 010 16h18a4 4 0 01-3.2-1.7l-4.57-6.54c-.3-.43-.76-.64-1.23-.64z" fill="white"/></svg>');
+        transition: opacity .3s ease, .3s visibility;
+        transition-delay: 16ms; /* delay one animation frame for better sync with tooltipBox */
         visibility: hidden;
         pointer-events: visible;
+        z-index: 2; /* move above tooltip-box's box-shadow */
     }
 
-    .tooltip > a.top::after {
-        border-color: var(--nimiq-blue-darkened) transparent transparent transparent;
+    .inverse-theme .trigger::after {
+        background: white;
+    }
+
+    .transition-position .trigger::after {
+        transition: top .2s ease, transform .2s ease, opacity .3s ease, .3s visibility;
+    }
+
+    .top .trigger::after {
         top: -2rem;
-        bottom: 0;
+        transform: scaleY(-1);
     }
 
-    .tooltip > a.bottom::after {
-        border-color: transparent transparent var(--nimiq-blue-darkened) transparent;
-        top: 0;
-        bottom: -2rem;
+    .bottom .trigger::after {
+        top: 100%;
     }
 
-    .tooltip a:focus::after,
-    .tooltip a:hover::after,
-    .tooltip.active > a::after {
+    .shown .trigger::after {
         opacity: 1;
         visibility: visible;
     }
 
     .tooltip-box {
         position: absolute;
-        visibility: hidden;
-        pointer-events: visible;
         color: white;
         background: var(--nimiq-blue-bg);
         padding: 1.5rem;
         border-radius: .5rem;
-        transition: opacity .3s ease, transform .2s ease, top .2s ease;
-        opacity: 0;
+        font-size: 1.75rem;
+        line-height: 1.5;
+        font-weight: 600;
+        transition: opacity .3s ease;
         box-shadow: 0 1.125rem 2.275rem rgba(0, 0, 0, 0.11);
+        z-index: 1;
```

Comment:
I think we can use much bigger z-index values for the tooltip, as I would expect it should always be on top. So something like `z-index: 999998` makes sense I think.